### PR TITLE
ART-18936: Remove ec_status field from BigQuery schemas and code

### PR DESCRIPTION
**Pull request by locriandev**
The ec_status field is being deprecated from BigQuery schemas. This commit removes:
- ec_status from KonfluxRecord EXCLUDED_KEYS
- ec_status parameter and field from all build record classes (KonfluxBuildRecord, KonfluxBundleBuildRecord, KonfluxFbcBuildRecord)
- ec_status from ECVerificationResult dataclass
- All code that sets or reads ec_status
- All tests related to ec_status
- BigQuery schema definitions containing ec_status

** Testing:
[ironic-container-v4.20.0-202605191147.p2.g4b0fc45.assembly.stream.el9](https://art-build-history-art-build-history.apps.artc2023.pc3z.p1.openshiftapps.com/build?nvr=ironic-container-v4.20.0-202605191147.p2.g4b0fc45.assembly.stream.el9&record_id=d6c27529-246c-16d8-d465-50a784c39563&group=openshift-4.20&outcome=success&type=image) was built from [this Jenkins job](https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/aos-cd-builds/job/build%252Focp4-konflux/34947/)

### CODE REVIEW

**Comment by locriandev** [comment]
/hold

Needs [ART-18937](https://redhat.atlassian.net/browse/ART-18937) to display the ITS status properly

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 1]
```diff
@@ -406,7 +397,7 @@ async def build(self, metadata: ImageMetadata, git_auth_secret: Optional[str] =
                         pipelinerun_name,
                         pipelinerun_info.to_dict(),
                     )
-                    if ec_failed:
+                    if ec_pipeline_url:
```
Comment (lgarciaaco) [inline doozer/doozerlib/backend/konflux_image_builder.py:411]: If EC throws before we get a PLR URL, `ec_pipeline_url` stays empty but `ec_result.ec_failed` is still true. Retries won't stop and pyartcd won't count it as EC. Old code keyed off `ec_failed` for both. Woud switching the retry check back to `ec_result.ec_failed` work, or should we set something on the record for exceptions?

File: artcommon/artcommonlib/konflux/konflux_build_record.py [hunk 2]
```diff
@@ -330,11 +328,6 @@ def __init__(
         self.embargoed = embargoed
         self.hermetic = hermetic
         self.artifact_type = artifact_type if isinstance(artifact_type, ArtifactType) else ArtifactType(artifact_type)
-        self.ec_status = (
-            ec_status
-            if isinstance(ec_status, KonfluxECStatus)
-            else KonfluxECStatus(ec_status or KonfluxECStatus.NOT_APPLICABLE.value)
-        )
         self.ec_pipeline_url = ec_pipeline_url
```
Comment (lgarciaaco) [inline artcommon/artcommonlib/konflux/konflux_build_record.py:365]: We dropped `ec_status` from the record but `KonfluxECStatus` is still defined above (~line 47) and nothing references it on this branch. Worth deleting the enum in the same PR, or keeping it for a follow-up?

File: doozer/tests/backend/test_konflux_ec_verification.py [hunk 3]
```diff
@@ -83,15 +83,11 @@ def _make_metadata(distgit_key="test-image", for_release=True, is_base_image=Fal
 
 
 def _ec_passed_result():
-    return ECVerificationResult(
-        ec_status=KonfluxECStatus.PASSED, ec_pipeline_url="https://example.com/ec-plr", ec_failed=False
-    )
+    return ECVerificationResult(ec_pipeline_url="https://example.com/ec-plr", ec_failed=False)
 
 
 def _ec_failed_result():
```
Comment (lgarciaaco) [inline doozer/tests/backend/test_konflux_ec_verification.py:89]: `_ec_failed_result()` always includes a PLR URL. Nothing here tests the exception path where `ec_failed=True` but `ec_pipeline_url` is empty. That's the case where retries keep going and pyartcd skips the EC bucket. Mind adding a helper + test (maybe near `test_no_retry_when_ec_fails`) that covers raise / empty URL and asserts we still break out and set the record correctly?

**Comment by lgarciaaco** [comment]
Re-reviewed at 2be55645, your replies on the inline threads look good (enum removal, conditional `ec_pipeline_url`, exception-path test).

/lgtm

File: doozer/doozerlib/backend/konflux_image_builder.py (hunk 1 above)
Comment (lgarciaaco) [reply doozer/doozerlib/backend/konflux_image_builder.py:411]: The 2be55645 approach works for me: keep the retry break on non-empty `ec_pipeline_url` (real ITS/policy failure with a PLR), and let infra/exception failures (`ec_failed` but no URL) retry. That matches the new test.

One follow-up for the branch: the legacy pyartcd counter tests still need `outcome` on record log entries (or a fallback in `update_build_fail_counters`). CI is failing on those three tests right now.
